### PR TITLE
Remeber Resolution + Faster Playlist + Empty Comment

### LINKED ISSUE #310: Faster Playlist Validation
Thread start from: https://github.com/NickvisionApps/TubeConverter/issues/129#issuecomment-1570375362

### LINKED ISSUE #326: Make`Comment` Metadata optional
Hello, when selecting "include metadata", the file is being "earmarked" by including a link to the original video in the "Comment" field. I was wondering if it would be possible to make this `Comment` metada optional? 

I also think that it would be useful to include a subtitle to describe what medata is being embedded:
image

My suggestion would be to have "Include link to original file" as a separate Gtk.Switch and have a Subtitles in the current switch to explain that the medata embedded contains Audio and Video properties only.

### LINKED ISSUE #332: [feature suggestion] Preset download quality
I often use this application to download videos, but only in 1280*720 quality (to avoid using too much bandwidth). It would be a good idea to allow the user to define in the settings (for example, in the "advanced options") a download quality that would be used by default, if available…

### PR DESCRIPTION
**Pull request by nlogozzo**
- [x] Closes #332
- [x] Closes #326
- [x] Closes #310

### CODE REVIEW

**Comment by nlogozzo** [comment]
With https://github.com/NickvisionApps/TubeConverter/pull/335/commits/43db19ebbaa0188ac86db365cef79eca603932a7, playlist validation now takes about the same amount of time as an individual video, no matter how may videos in the playlist. Take [this 50 video playlist](https://www.youtube.com/watch?v=vYp14UesizY&list=PLzCxunOM5WFIVeCUZ7bFCW7wwA_JBHsDC):

[Screencast from 2023-06-07 20-41-06.webm](https://github.com/NickvisionApps/TubeConverter/assets/17648453/4168121b-fa8b-4167-b633-db6c4afa5925)

File: NickvisionTubeConverter.Shared/Models/MediaUrlInfo.cs [hunk 1]
```diff
@@ -107,40 +112,72 @@ private MediaUrlInfo(string url)
...
+    /// <param name="defaultUrl">A default URL to use if none available</param>
+    private void ParseFromPyDict(dynamic yt, PyDict mediaInfo, bool isPartOfPlaylist, string defaultUrl)
     {
         var title = mediaInfo.HasKey("title") ? (mediaInfo["title"].As<string?>() ?? "Media") : "Media";
         foreach (var c in Path.GetInvalidFileNameChars())
         {
             title = title.Replace(c, '_');
         }
-        foreach (var f in mediaInfo["formats"].As<PyList>())
+        if(mediaInfo.HasKey("formats"))
```
Comment (DaPigGuy) [inline NickvisionTubeConverter.Shared/Models/MediaUrlInfo.cs:152]: This condition is repeated in `ParseFormats`

**Review by DaPigGuy — CHANGES_REQUESTED** [review]

**Comment by fsobolev** [comment]
image

Something is broken somewhere...

File: NickvisionTubeConverter.Shared/Models/MediaUrlInfo.cs (hunk 1 above)
Comment (fsobolev) [reply NickvisionTubeConverter.Shared/Models/MediaUrlInfo.cs:149]: Fixed in d0392cdcd02e3e94c213cdd417f23ba8f1b1f1a2

**Pull request merged**